app/bundles/LeadBundle/Entity/LeadListRepository.php

```diff
@@ -448,6 +473,26 @@ public function getLeadsByList($lists, $args = array())
         return $return;
     }
 
+    public function arrangeFilters($filters) {
+        $objectFilters = [];
+
+        foreach ($filters as $filter) {
+            $object = (isset($filter['object'])) ? $filter['object'] : 'lead';
+            switch ($object) {
+                case 'lead' :
```

Comment:
Its being used to separate the filters for leads table and companies table, otherwise they get mixed up and fields are being queried in both tables.